addons/fetchmail/models/fetchmail.py

```diff
@@ -17,6 +18,10 @@
 # Workaround for Python 2.7.8 bug https://bugs.python.org/issue23906
 poplib._MAXLINE = 65536
 
+# Add timeout to IMAP connections
+# HACK https://bugs.python.org/issue38615
+IMAP4._create_socket = lambda self: socket.create_connection((self.host or None, self.port), MAIL_TIMEOUT)
```

Comment:
Can't you give the timeout to the `IMAP4` constructor ? (in the link you added, they say that they merged the fix, if it's only for > python 3.9, maybe add a comment that it would be better to give it to the `IMAP4` constructor, but that required python 3.9, so when we will change the version, we can clean this)

Reply:
Yes, the current patch would make 3.9 crash due to the change introduced by bpo-38615, no? https://github.com/python/cpython/commit/13a7ee8d62dafe7d2291708312fa2a86e171c7fa

Could we update it in a way that works for 3.9 *and* earlier, and add a comment to drop this later and fix the IMAP4 construction?
```suggestion
IMAP4._create_socket = lambda self, timeout=MAIL_TIMEOUT: socket.create_connection((self.host or None, self.port), timeout)
```